### ISSUE TITLE
statistics plugin month/day "spectrum" broken?

### ISSUE THREAD

**Issue by ophian** [s9y/Serendipity#273]
Someone has "broken"(?) the statistics plugin spectrum analyzer visualization for day / month visits.
![statistic-spectrum](https://cloud.githubusercontent.com/assets/1105372/5887137/05400240-a3c1-11e4-861d-30b6a0ba4ce3.png)
Or does it not set steps right for low visited sites? (I even remember to have seen this on netzgestaltung)

**Comment by ophian** [comment]
yeah on netzgestaltung it is the same - but the page all over is looking quite different....
![statistic-spectrum-ng](https://cloud.githubusercontent.com/assets/1105372/5887284/82289424-a3c7-11e4-8c45-3cada3b727cb.png)

It even seems someone has deleted the statistics plugin at all - that is why it looks so different on our dev blog! It currently only lives in the 1.7 branch, as v. 1.53 - that is bad!

**Comment by ophian** [comment]
...and as a new styled plugin with the "old" version number on netzgestaltung. So keep it, @yellowled !

**Comment by yellowled** [comment]
Apparently, it was removed in 37b55759f24310edc96337424437ca14c31c8f81. That's the commit cleaning up the core plugin selection, so it was supposed to be moved to s9y/additional_plugins.

So either it got lost in the move or it was never supposed to be moved.

**Comment by yellowled** [comment]
> So keep it, @yellowled !

Are you aware that this is *exactly* why we use a version control system? I may not know *how* just yet, but I'm pretty sure we can just restore the version from before 37b55759f24310edc96337424437ca14c31c8f81.

**Comment by ophian** [comment]
Yes I am, but it seems pretty much easier using the one from ng and upload it with a new version num.

**Comment by yellowled** [comment]
Considering https://github.com/s9y/Serendipity/issues/71#issuecomment-61411021, I'd say it was never supposed to be moved to additional_plugins (see http://board.s9y.org/viewtopic.php?p=10439527#p10439527 for Garvin's list of plugins).

> Yes I am, but it seems pretty much easier using the one from ng and upload it with a new version num.

The one on nbe might have been edited at some point, it might have changes that I never commited. I think it would be safer to restore an older version from git. (Still don't know how to do that, though.)

**Comment by yellowled** [comment]
In any case, I'd want to wait on @onli and @garvinhicking to chime in on this, maybe @ddeimeke can help out with the git stuff?

**Comment by onli** [comment]
Huh, possible I made a mistake there. It really is missing in additional_plugins and the 2.0 branch?

To fix this the git way, to my knowledge we'd have to revert the commit and then do it again without the statistics plugin. That is not too complicated, but also not really nice. It would be easier to just keep it outside the core and add it to additional_plugins. I'm actually quite happy to not have a plugin in the core we recommend anyway to not use - it needs a rework first to not slow down the blog after some time. And till then it should imo not be bundled by default.

**Comment by yellowled** [comment]
I'm fine with not having it in the core. We usually tell people in the forums to use Piwik or Analytics for better results anyway. However, it should still be moved to s9y/additional_plugins soon-ish so the people who currently use it can update, right? @garvinhicking, @ophian?

@onli, can you handle the move to s9y/additional_plugins from the version before 37b55759f24310edc96337424437ca14c31c8f81?

**Comment by ophian** [comment]
If you zip and send me the full nbe version, I am going to fix this for tomorrow,
A 
```css
.serendipity_statistic img
.coll img {
    max-height: inherit !important;
}
```
should fix the missing spectrum height with 2.0 versions. (I could not check if the important is really necessary though)
And @onli  should please check his commits, if that was the only plugin getting lost.

**Comment by yellowled** [comment]
I backed up the nbe version to http://7ax.de/7e3i – however, I would still prefer it if these were restored from GitHub.

**Comment by garvinhicking** [comment]
I'm fine with moving the statistics plugin outside core, think that was already planned in first place, so no worries from me.

**Comment by ophian** [comment]
If anyone is worried about loosing history, this is true, but even the other ones moved from core to additional_plugins are new from the first commit to the new place, see https://github.com/s9y/additional_plugins/commits/master/serendipity_event_karma/serendipity_event_karma.php

**Comment by ophian** [comment]
I think all the history got lost (well, for the new place only) for all these plugins is, because they were erased and then new committed and not just being moved to the new place. Git should know about moving commits.

If any one wants this to be done, we would need to reset the adding commit https://github.com/s9y/additional_plugins/commit/c18e793b70a70f217e529ee59f2111709dad99f9 and reset the https://github.com/s9y/Serendipity/commit/37b55759f24310edc96337424437ca14c31c8f81 erasing one too. Then move them altogether to the new place, commit the changes to plugin_api, re-add the statics changes, and all other which might have been commit since then, etc.

Another thing I noticed: Since being core plugins before, they only used `@serendipity_plugin_api::load_language(dirname(__FILE__));` and no `if (IN_serendipity !== true) { die ("Don't hack!"); }`. I did not check them alltogether, but it seems, they were not set to require 2.0 or 1.7 only.  What shall we do about it?

@garvinhicking @onli
